### PR TITLE
[feat] Use cryptography instead of pycryptopp to reduce dependencies.

### PR DESCRIPTION
**Pull request by meskio**
* Resolves: #7889

### CODE REVIEW

File: src/leap/mail/walk.py [hunk 1]
```diff
@@ -19,18 +19,24 @@
...
+
+def _get_hash(s):
+    digest = hashes.Hash(hashes.SHA256(), default_backend())
+    digest.update(s)
+    return digest.finalize().encode("hex").upper()
+
 if DEBUG:
     def get_hash(s):
-        return sha256.SHA256(s).hexdigest()[:10]
+        return _get_hash(s)[:10]
```
Comment (meskio) [inline src/leap/mail/walk.py]: Why there is this debug function?

Comment (kalikaneko) [inline src/leap/mail/walk.py]: it's a remnant from when this module was in development, it shows and print different parts of the message.

Comment (meskio) [inline src/leap/mail/walk.py]: Can we remove it?

Comment (kalikaneko) [inline src/leap/mail/walk.py]: I'm ok with removing it, do you mind to include it in the pr? 
note there're several refs to DEBUG in the module

Comment (meskio) [inline src/leap/mail/walk.py]: Sure, let me update it.

**Pull request merged**